includes/class-wc-stripe-intent-controller.php

```diff
@@ -1101,7 +1101,7 @@ public function confirm_change_payment_from_setup_intent_ajax() {
 				throw new WC_Stripe_Exception( 'subscription_not_found', __( "We're not able to process this subscription change payment request payment. Please try again later.", 'woocommerce-gateway-stripe' ) );
 			}
 
-			$setup_intent_id = isset( $_POST['intent_id'] ) ? wc_clean( wp_unslash( $_POST['intent_id'] ) ) : null;
+			$setup_intent_id = ( isset( $_POST['intent_id'] ) && is_string( $_POST['intent_id'] ) ) ? sanitize_text_field( wp_unslash( $_POST['intent_id'] ) ) : null;
```

Comment:
I think this is fine. We do expect a valid `$_POST['intent_id']` to be a string.